src/Bicep.Core/Emit/ResourceDependencyVisitor.cs

```diff
@@ -24,7 +25,8 @@ public class ResourceDependencyVisitor : SyntaxVisitor
             {
                 if (kvp.Key is ResourceSymbol resourceSymbol)
                 {
-                    output[resourceSymbol] = kvp.Value.ToImmutableHashSet();
+                    var ancestors = model.ResourceAncestors.GetAncestors(resourceSymbol);
+                    output[resourceSymbol] = kvp.Value.Concat(ancestors).ToImmutableHashSet();
```

Comment:
Ancestors are automatically a dependency.

Reply:
If we generate using the child resource syntax in the template, this is actually required :)

---
In reply to: [577215124](https://github.com/Azure/bicep/pull/1516#discussion_r577215124) [](ancestors = 577215124)

Reply:
Does this mean all ancestors are regarded as a dependency? Technically it should only be necessary to declare the direct parent in the generated "dependsOn" list, and this is normally how it would be handwritten.

Not super important as we don't care too much about the verbosity of the generated JSON.

Reply:
> Does this mean all ancestors are regarded as a dependency? 

That's how I implemented it. I think this actually might be needed for correctness.

Think about a case like this:

```
resource A '....' = {
  resource B '....' existing = {
    resource C '....' = {
    }
  }
}
```

Adding all ancestors to `dependsOn` will guarantee that modifications made to `A` are made before modifications made to `C`. That's a little subtle, but that's the semantic I'd expect. Recall that `existing` resources will not be added to the dependency list.

Reply:
Right, I guess I meant "closest deployable parent" rather than "direct parent".

Reply:
So the code currently includes all ancestors, but I don't have a strong reason to change it or leave it 😆 

I'm going to resolve this. LMK if you have a decision to change it 👍